webapp/sdf/src/components/item-component/item-component.spec.js

```diff
@@ -0,0 +1,61 @@
+import { mount } from '@vue/test-utils'
+import ItemComponent from 'components/item-component/item-component.vue'
+import Item from 'classes/item.js'
+
+describe('item-component', () => {
+let component
+
+    beforeEach(() => {
+        component = mount(ItemComponent, {
+            propsData: {
+                item : new Item(1, "Eintrag2", 2)
+            }
+        });
+    })
+
+    it('contains item title and votes Expected --> Eintrag (2)', () => {
+        expect(component.text()).toContain('Eintrag2 (2)')
+    })
+
+    it('emits remove event', async () => {
+        const removeButton = component.find('.remove-button')
+        removeButton.trigger('click')
+        await component.vm.$nextTick()
+        expect(component.emitted().removeEvent).toBeTruthy()
+    })
+
+    it('emits update event +1', async () => {
+        const upvoteButton = component.find('.upvote-button')
+        upvoteButton.trigger('click')
+        await component.vm.$nextTick()
+        expect(component.emitted().updateEvent).toBeTruthy()
+    })
+
+    it('emits update event -1', async () => {
+        const downvoteButton = component.find('.downvote-button')
+        downvoteButton.trigger('click')
+        await component.vm.$nextTick()
+        expect(component.emitted().updateEvent).toBeTruthy()
+    })
```

Comment:
Test case description and test case implementation do not match. You would need to assert this:
```suggestion
    it('emits update event +1', async () => {
        const upvoteButton = component.find('.upvote-button')
        upvoteButton.trigger('click')
        await component.vm.$nextTick()
        expect(component.emitted('updateEvent').toMatchObject({votes: 3})
    })

    it('emits update event -1', async () => {
        const downvoteButton = component.find('.downvote-button')
        downvoteButton.trigger('click')
        await component.vm.$nextTick()
        expect(component.emitted('updateEvent').toMatchObject({votes: 1})
    })
```